src/runtime/virtcontainers/clh.go

```diff
@@ -882,16 +966,64 @@ func (clh *cloudHypervisor) Cleanup(ctx context.Context) error {
 
 func (clh *cloudHypervisor) PauseVM(ctx context.Context) error {
 	clh.Logger().WithField("function", "PauseVM").Info("Pause Sandbox")
+
+	cl := clh.client()
+
+	_, err := cl.PauseVM(ctx)
+	if err != nil {
+		return openAPIClientError(err)
+	}
+
 	return nil
 }
 
 func (clh *cloudHypervisor) SaveVM() error {
 	clh.Logger().WithField("function", "saveSandboxC").Info("Save Sandbox")
+
+	cl := clh.client()
+
+	_, err := clh.vmInfo()
+	if err != nil {
+		clh.Logger().WithField("function", "SaveVM").WithError(err).Info("[clh] vmInfo failed")
+		return openAPIClientError(err)
+	}
+
+	snaopshotConfig := *chclient.NewVmSnapshotConfig()
+	destinationUrl := "file://" + filepath.Dir(clh.config.DevicesStatePath)
```

Comment:
Do you need to check that `clh.config.DevicesStatePath` has a value?

Reply:
Thanks for your review, DevicesStatePath has been checked in function CheckTemplateConfig() of virtcontainers/hypervisor.go.

Reply:
Ok, but how would I (and this function) know that? Is it possible to call this function without `CheckTemplateConfig()` being called? Even if it isn't, that is also not clear from this point in the code.